pkg/objectstore/bucket.go

```diff
@@ -193,6 +194,27 @@ func s3BucketRegion(ctx context.Context, cfg ProviderConfig, sec Secret, bucketN
 		return "", errors.Wrapf(err, "failed to create session, region = %s", r)
 	}
 	svc := s3.New(s)
+
+	// s3-compatible stores may not support s3manager.GetBucketRegion() API, so
+	// prefer to use the get-bucket-location API instead
+	if cfg.Endpoint != "" {
+		if resp, err := svc.GetBucketLocation(&s3.GetBucketLocationInput{
+			Bucket: aws.String(bucketName),
+		}); err == nil {
+			if resp.LocationConstraint == nil { // per the AWS SDK doc a nil location means us-east-1
+				return "us-east-1", nil
+			}
+			return *resp.LocationConstraint, nil
+		} else {
+			log.Error().
+				WithContext(ctx).
+				WithError(err).
+				Print("GetBucketLocation() failed, falling back to GetBucketRegion()", field.M{"config": c})
+		}
```

Comment:
[nit] I think it's slightly more readable to separate this out into multiple lines
```suggestion
		resp, err := svc.GetBucketLocation(&s3.GetBucketLocationInput{
			Bucket: aws.String(bucketName),
		})
		if err == nil {
			if resp.LocationConstraint == nil { // per the AWS SDK doc a nil location means us-east-1
				return "us-east-1", nil
			}
			return *resp.LocationConstraint, nil
		}
		log.Error().
			WithContext(ctx).
			WithError(err).
			Print("GetBucketLocation() failed, falling back to GetBucketRegion()", field.M{"config": c})
```